### PR TITLE
refactor: use whiskers & adhere to palette

### PR DESCRIPTION
**Pull request by uncenter**
Adds [Whiskers](https://github.com/catppuccin/toolbox/tree/main/whiskers), Catppuccin's internal templating tool, to build the themes. Instead of editing the `.json` files directly, edit the `element.tera` file and then run `just build` to build the output themes. You will need Whiskers (linked above) and *optionally* [Just](https://github.com/casey/just) installed.

~~Note: currently a draft waiting on https://github.com/catppuccin/toolbox/pull/217.~~

### CODE REVIEW

File: templates/element.tera [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  "is_dark": {{ flavor.dark }},
+  "colors": {
+    "accent-color": "#{{ palette[accent].hex }}",
+    "primary-color": "#{{ palette[accent].hex }}",
+    "warning-color": "#{{ palette.red.hex }}",
+    "alert": "#{{ palette.yellow.hex }}",
+    "sidebar-color": "#{{ palette.crust.hex }}",
+    "roomlist-background-color": "#{{ palette.mantle.hex }}",
+    "roomlist-text-color": "#{{ palette.text.hex }}",
+    "roomlist-text-secondary-color": "#{{ palette.base.hex }}",
```
Comment (isabelroses) [inline templates/element.tera:23]: Do you mind fixing
https://github.com/catppuccin/element/issues/4

Comment (uncenter) [reply templates/element.tera:23]: Separate PR I think. This just aims to refactor the build system atm.

**Review by sgoudham — APPROVED** [review]

File: templates/element.tera (hunk 1 above)
Comment (uncenter) [reply templates/element.tera:23]: @isabelroses do you want to make the follow up PR or shall I?

Comment (isabelroses) [reply templates/element.tera:23]: I don't mind doing it

Comment (uncenter) [reply templates/element.tera:23]: Also might be worth getting rid of unnecessary `palette` usage now that v2.3.0 is out

**Pull request merged**